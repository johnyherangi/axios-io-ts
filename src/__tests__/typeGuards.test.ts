import { onDecodeError } from "@src/typeGuards"

describe("typeGuards.ts", () => {
    describe("onDecodeError", () => {
        it("captures a DecodeError", async () => {
            const result = await Promise.reject({ url: "/test", errors: ["test error"] }).catch(
                onDecodeError(() => "decode error"),
            )
            expect(result).toMatchSnapshot()
        })
        it("rethrows an unknown error", async () => {
            const result = await Promise.reject("unknown error")
                .catch(onDecodeError(() => "decode error"))
                .catch((err) => err)
            expect(result).toMatchSnapshot()
        })
    })
})
