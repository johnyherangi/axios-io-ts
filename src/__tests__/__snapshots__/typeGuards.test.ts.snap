// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeGuards.ts onDecodeError captures a DecodeError 1`] = `"decode error"`;

exports[`typeGuards.ts onDecodeError rethrows an unknown error 1`] = `"unknown error"`;
